File: 2023/go/day00/basics.go
package main

import (
	"os"
	"strconv"
	"strings"
)

func main() {
	data, _ := os.ReadFile("input.txt")
	lines := strings.Split(string(data), "\n")
	for _, line := range lines {
		handleLine(line)
	}
}

func handleLine(line string) {
	numbers := strings.Split(line, " ")
	sum := int(0)
	for _, number := range numbers {
		number, _ := strconv.ParseInt(number, 10, 64)
		sum += int(number)
	}
	println(sum)
}

File: 2023/elixir/day00/basics.exs
input = File.read!("input.txt")
lines = String.split(input, "\n")

Enum.map(lines, fn l ->
  if l != "" do
    numbers = String.split(l, " ")

    sum =
      List.foldl(numbers, 0, fn n, acc ->
        case Integer.parse(n) do
          {i, _} -> i + acc
          _ -> 0
        end
      end)

    IO.puts(sum)
  end
end)
